cargo-cyclonedx/src/generator.rs

```diff
@@ -542,6 +593,45 @@ fn index_resolve(packages: Vec<Node>) -> ResolveMap {
         .collect()
 }
 
+fn index_dep_kinds(root: &PackageId, resolve: &ResolveMap) -> DependencyKindMap {
+    // cache strongest found dependency kind for every node
+    let mut id_to_dep_kind: HashMap<&PackageId, PrivateDepKind> = HashMap::new();
+    id_to_dep_kind.insert(root, PrivateDepKind::Runtime);
+
+    let root_node = &resolve[root];
+    let mut nodes_to_visit = LinkedList::<&Node>::new();
```

Comment:
Probably not relevant for performance in the grand scheme of things, but still: why a linked list? It is almost never the right tool for the job. Wouldn't a simple `Vec` where you `push` and `pop` be enough? That's semantically equivalent to `push_front` + `pop_front` on a linked list.

Reply:
Good point. Will change to a `Vec`.